src/main/java/com/example/demo/services/UsersService.java

```diff
@@ -125,12 +141,382 @@ public Integer extractData(ResultSet rs) throws SQLException, DataAccessExceptio
+			
+			@Override
+			public List<SalaryDetailsBO> extractData(ResultSet rs) throws SQLException, DataAccessException {
+				
+				List<SalaryDetailsBO> salaryDetailsBO = new ArrayList(); 
+				while (rs.next()) 
+				{
+					SalaryDetailsBO salaryDet = new SalaryDetailsBO();
+					salaryDet.setAmount(rs.getDouble("amount"));
+					salaryDet.setEffDate(rs.getDate("eff_date"));
+					salaryDet.setEndDate(rs.getDate("end_date"));
+					
+					salaryDetailsBO.add(salaryDet);
+				}
+				return salaryDetailsBO;
+			}
+			
+		});
+	}
+	
+	//Prototype to get Payroll details in user_payroll_details
+	//private Map<String, PayrollDetailsBO> userPayrollDetails; // Map of Academic Id to Payroll details
+	public Map<String, PayrollDetailsBO> getPayrollDetailsOfOneUser(String userId)
+	{
+		String query = "SELECT * FROM USER_PAYROLL_DETAILS where user_id=?";
+		return jdbcTemplate.query(query, new PreparedStatementSetter() {
+			
+			@Override
+			public void setValues(PreparedStatement ps) throws SQLException {
+				ps.setString(1, userId);
+			}
+			
+		}, new ResultSetExtractor<Map<String, PayrollDetailsBO>>(){
+			
+			@Override
+			public Map<String, PayrollDetailsBO> extractData(ResultSet rs) throws SQLException, DataAccessException {
+				Map<String, PayrollDetailsBO> userPayrollDetailsMap = new HashMap<>();
+				while(rs.next())
+				{
+					String academicID = rs.getString("ACADEMIC_ID");
+					
+					
+					if(userPayrollDetailsMap.containsKey(academicID))
+					{
+						//As of now, not considering this. But there is a problem here.
+						//When the AcademicID is existing in the map, and then if we try to add another, it will override the previous one.
+						//May be List should be there such as
+						//  Map<String, List<PayrollDetailsBO>>
+					}
+					else
+					{
+						
+						PayrollDetailsBO payrollDetails = new PayrollDetailsBO();
+						payrollDetails.setPayrollDate(rs.getDate("payroll_date"));
+						payrollDetails.setPaidDays(rs.getDouble("paid_days"));
+						payrollDetails.setUnpaidDays(rs.getDouble("unpaid_day"));
+						payrollDetails.setAmount(rs.getDouble("amount"));
+						payrollDetails.setPayrollLocked(rs.getString("payroll_locked").charAt(0));
+						
+						int month_id = rs.getInt("month_id");
+						String monthID = ""+month_id;
+						String month = getMonthNameFromMonthMaster(monthID); 
+						
+						
+						MonthMasterDetailsBO monthMasterDetailsBO = new MonthMasterDetailsBO();
+						monthMasterDetailsBO.setMonthId(month_id);
+						monthMasterDetailsBO.setMonthName(month);
+						
+						payrollDetails.setPayrollMonth(monthMasterDetailsBO);
+						
+						
+						
+						
+						userPayrollDetailsMap.put(academicID, payrollDetails);
+					}
+					
+				}
+				return userPayrollDetailsMap;
+			}
+		});
+	}
+	
+	public UserAdvanceDetailsBO getUsersAdvanceDetails(String userId)
+	{
+		
+		populateInheritedEntriesofUserAdvanceDetailsBO(getBasicDetailsOfOneUser(userId));
```

Comment:
remove populateInheritedEntriesofUserAdvanceDetailsBO add everythimg in the getUserBasicDetails